ambariclient/models.py

```diff
@@ -887,12 +887,23 @@ def execute_action(self, action, context, parameters=None, hosts=None):
         return self.request
 
     def decommission(self, service, hosts):
-        """Decommission slave components on a cluster.
+        self.commission(service, hosts, "decommission")
+
+    def recommission(self, service, hosts):
+        self.commission(service, hosts, "recommission")
+        """Nodemanager recommission does not start nodemanager process on the host. so its required to do it separately"""
+        if service is "YARN":
```

Comment:
Isn't the recommission an asynchronous process?  Shouldn't you wait until the request is finished to submit this `start()` request?  Or is Ambari smart enough to queue it up and wait for the recommission to complete?